Size embeds using aspect-ratio, remove fixed-size options

eyeseast opened the pull request:
Closes #1119 

View live embeds from this branch on our embed test page: https://muckrock.github.io/documentcloud-embed-examples/

I added a minimalist resize script to notes and pages.

eyeseast commented on src/lib/api/embed.ts:
```diff
@@ -13,13 +16,55 @@ export function embedUrl(url: URL | string): URL {
+  const height = page_size[1] ?? 11;
+
+  const note_width = note ? notes.width(note) * width : null;
+  const note_height = note ? notes.height(note) * height : null;
```
I've thought about just doubling the height of the highlight, as a baseline. Though this is where a resizer script would help us the most.

eyeseast replied:
I added a resize script and it's much better.

allanlasser commented:
Agree that the resizer script works great—feels very similar to what we had before with the `enhance.js` script.

https://github.com/user-attachments/assets/c38bf868-c6d6-4cd0-b0e0-662a76a3427c

I noticed some behavior on the test site where the resizer didn't always apply to the embed. Perhaps there's some rendering that's happening slower than the script expects?

One other small bit of feedback is on the script name itself. If this will be loaded on third-party sites, it may benefit from a more descriptive name (`resize-documentcloud-embed-resize.js`?) so that users and site owners know where the script is coming from and what it's doing.

allanlasser commented on src/lib/api/embed.ts:
```diff
@@ -13,13 +16,55 @@ export function embedUrl(url: URL | string): URL {
+    : [[8.5, 11]];
+  const page_size = sizes[page - 1] ?? [];
+  const width = page_size[0] ?? 8.5;
+  const height = page_size[1] ?? 11;
```
Since the resizer script could be blocked by third-party content blocking rules, I think a percentage-based pad could be a good naive fallback.

allanlasser reviewed (commented):
Looking good!

allanlasser commented on src/embed/dc-resize.js:
```diff
@@ -0,0 +1,21 @@
+  if (![data.width, data.height, data.href].every(Boolean)) return;
+
+  // find all the iframe with this href
+  const iframes = document.querySelectorAll(`iframe[src*="${data.href}"]`);
```
This contains selector could lead to bugs through unlikely, but possible partial matches (e.g. `/document/123` would also match `/document/1234`). A more robust solution would be URL comparison:
```
const iframes = Array.from(document.querySelectorAll(`iframe`)).filter(iframe => {
  try {
    const iframeSrc = new URL(iframe.src);
    const dataSrc = new URL(data.href);
    return iframeSrc.path === dataSrc.path;
    // could go further with search string comparison
    // && iframeSrc.search === dataSrc.search;
  } catch {
    return iframe.src === data.href;
  }
})
```

eyeseast commented:
I renamed the resize script to `dc-resize.js`. I figure it will already have our domain attached to it, too.

In a previous version of this PR, the note embed was resizing a little bit every second, causing the iframe to resize in a stuttering way, so I added an option to turn off the event listener for `onresize`. I'm erring on the side of having some embeds be too small instead of stuttering.
